zig-dxread/src/main.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const fs = std.fs;
+const gpa = std.heap.GeneralPurposeAllocator(.{}){};
+const allocator = gpa.allocator;
+const ArrayList = std.ArrayList;
+const erorset_1 = error{OutOfMemory};
+
+const RECORD_SIZE = 80 * 384;
+
+const DX_header = struct {
+    year: u32,
+    month: u32,
+    day: u32,
+    utc: u32,
+    satid: u32,
+    sattyp: u32,
+    nchans: u32,
+    nghtimg: u32,
+};
+
+const DX_data = struct {
+    header: DX_header,
+};
+
+
+pub fn readHeader(buffer: [RECORD_SIZE]u8) !DX_header {
+    // std.log.info("{any}", .{buffer});
+    std.log.info("value: {any}", .{buffer[10]});
+    var year: u32 = 1; //try std.fmt.parseInt(u32, buffer[0..8], 10);
+    return DX_header{
+        .year = year,
+        .month = 0,
+        .day = 0,
+        .utc = 0,
+        .satid = 0,
+        .sattyp = 0,
+        .nchans = 0,
+        .nghtimg = 0,
+    };
+}
+
+pub fn readFile(filename: [] const u8) !DX_data {
+    var file = try std.fs.openFileAbsolute(filename, .{});
+    defer file.close();
+
+    var buf_reader = std.io.bufferedReader(file.reader());
+    var in_stream = buf_reader.reader();
+
+    var buf: [RECORD_SIZE]u8 = try in_stream.readBytesNoEof(RECORD_SIZE);
+    std.log.info("First few bytes: {any}", .{buf[0..10]});
+    var header = try readHeader(buf);
+    std.log.info(" value: {any}", .{header});
+    return DX_data{.header = header};
+}
+
+pub fn makeList() ArrayList(u8) {
```

Comment:
This would be code-small in a zig library/application because the choice of allocator is set and may not be optimal; I understand that this may be test code but usually you construct the list and defer deinitialization where it's needed

Reply:
yeah this was just an experiment. i didnt mean to leave this here. i agree, the implicit assumption of an allocator at the lower level is definitely not a good idea.